Scripts
jupyter console --kernel octave
export PATH=$PATH:~/.local/bin 

# Atividade T1.1

**Seja $p(x) = ax^{3}+bx^{2}+cx+d$ o polinômio interpolador dos pontos abaixo:**

(−4,−10)
(−2,−3)
(2,−1)
(5,10)

**Determine o valor de b com pelo menos 4 casas decimais corretas.**



1. Vamos transformar o problema do polinômio interpolador em um sistema linear, com matriz ampliada:

$$
\begin{bmatrix}
-64 & 16 & -4 & 1\\
-8 & 4 & -2 & 1\\
8 & 4 & 2 & 1\\
125 & 25 & 5 & 1
\end{bmatrix} . \begin{bmatrix} 
a\\
b\\
c\\
d
\end{bmatrix} = \begin{bmatrix}
-10\\
-3\\
-1\\
10
\end{bmatrix}
$$

Extraindo a matriz ampliada temos:

$$
\begin{bmatrix}
-64 & 16 & -4 & 1 & -10\\
-8 & 4 & -2 & 1 & -3\\
8 & 4 & 2 & 1 & -1\\
125 & 25 & 5 & 1 & 10
\end{bmatrix}
$$

Precisamos escalonar essa matriz para extrairmos os valores das variáveis:

In [3]:
A = [-64 16 -4 1; -8 4 -2 1; 8 4 2 1; 125 25 5 1]
b = [-10; -3; -1; 10]

[x, A, b] = gaussPivotP(A, b)

A =

   -64    16    -4     1
    -8     4    -2     1
     8     4     2     1
   125    25     5     1

b =

  -10
   -3
   -1
   10

x =

   0.105820
  -0.076720
   0.076720
  -1.693122

A =

   125.0000    25.0000     5.0000     1.0000
          0    28.8000    -1.4400     1.5120
          0          0     1.8000     0.8100
          0          0          0     1.4000

b =

   10.0000
   -4.8800
   -1.2333
   -2.3704



### Resultado

Como vemos o resultado do escalonamento nos trás:

$$
\begin{bmatrix}
   125.0000  &  25.0000  &   5.0000  &   1.0000 & 10.0000\\
          0  &  28.8000    & -1.4400   &  1.5120 & -4.8800\\
          0  &        0    & 1.8000  &   0.8100 & -1.2333\\
          0  &        0    &      0  &   1.4000 & -2.3704
\end{bmatrix}
$$

Nos dando o resultado:

$$
x = 
\begin{bmatrix}
   0.105820\\
  -0.076720\\
   0.076720\\
  -1.693122
\end{bmatrix}
$$

Assim $b = -0.0767$.